src/libraries/System.Net.Quic/src/System.Net.Quic.csproj

```diff
@@ -1,11 +1,15 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
-    <TargetFrameworks>$(NetCoreAppCurrent)-windows;$(NetCoreAppCurrent)-Linux;$(NetCoreAppCurrent)-OSX;$(NetCoreAppCurrent)-FreeBSD;$(NetCoreAppCurrent)-iOS;$(NetCoreAppCurrent)-tvOS</TargetFrameworks>
+    <TargetFrameworks>$(NetCoreAppCurrent)-windows;$(NetCoreAppCurrent)-Linux;$(NetCoreAppCurrent)-OSX;$(NetCoreAppCurrent)-FreeBSD;$(NetCoreAppCurrent)-iOS;$(NetCoreAppCurrent)-tvOS;$(NetCoreAppCurrent)</TargetFrameworks>
```

Comment:
Why `$(NetCoreAppCurrent)` rather than `$(NetCoreAppCurrent)-Browser`?

Reply:
Ah, I see in your PR description:
"Adding a non-rid configuration so that this applies for all future configurations that aren't explicitely listed as well."
I see other projects use:
```
<GeneratePlatformNotSupportedAssemblyMessage Condition="'$(TargetsBrowser)' == 'true'">
```
Why is this one different?

Reply:
My gut feeling is that for any future configuration that doesn't derive from any of the specified TargetFrameworks we should default to produce a PNSE assembly (as that's required for inbox libraries). I don't know why we are not following this pattern elsewhere but I assume because there isn't any guidance on this written down yet.

Reply:
This PR is the one that appears to be introducing an inconsistency (if I'm wrong, just let me know that).  If that's true, then either a) this should follow the same pattern used elsewhere, or b) all other uses should be changed over as well.

Reply:
I don't have a strong opinion on the change and I'm fine with switching it to use a `Browser` configuration instead but I'm wondering if consistency or correctness is more important in this case.

Reply:
My point is, if Browser is actually incorrect, then all of the other ones are incorrect, too, and should be fixed.  If it's not incorrect, then we should be consistent.

```
d:\repos\runtime\src\libraries\System.Net.HttpListener\src\System.Net.HttpListener.csproj(7):    <GeneratePlatformNotSupportedAssemblyMessage Condition="'$(TargetsBrowser)' == 'true'">SR.SystemNetHttpListener_PlatformNotSupported</GeneratePlatformNotSupportedAssemblyMessage>
d:\repos\runtime\src\libraries\System.Net.NameResolution\src\System.Net.NameResolution.csproj(8):    <GeneratePlatformNotSupportedAssemblyMessage Condition="'$(TargetsBrowser)' == 'true'">SR.SystemNetNameResolution_PlatformNotSupported</GeneratePlatformNotSupportedAssemblyMessage>
d:\repos\runtime\src\libraries\System.Net.NetworkInformation\src\System.Net.NetworkInformation.csproj(8):    <GeneratePlatformNotSupportedAssemblyMessage Condition="'$(TargetsBrowser)' == 'true'">SR.SystemNetNetworkInformation_PlatformNotSupported</GeneratePlatformNotSupportedAssemblyMessage>
d:\repos\runtime\src\libraries\System.Net.Ping\src\System.Net.Ping.csproj(8):    <GeneratePlatformNotSupportedAssemblyMessage Condition="'$(TargetsBrowser)' == 'true'">SR.SystemNetPing_PlatformNotSupported</GeneratePlatformNotSupportedAssemblyMessage>
d:\repos\runtime\src\libraries\System.Net.Requests\src\System.Net.Requests.csproj(8):    <GeneratePlatformNotSupportedAssemblyMessage Condition="'$(TargetsBrowser)' == 'true'">SR.SystemNetRequests_PlatformNotSupported</GeneratePlatformNotSupportedAssemblyMessage>
d:\repos\runtime\src\libraries\System.Net.Security\src\System.Net.Security.csproj(10):    <GeneratePlatformNotSupportedAssemblyMessage Condition="'$(TargetsBrowser)' == 'true'">SR.SystemNetSecurity_PlatformNotSupported</GeneratePlatformNotSupportedAssemblyMessage>
d:\repos\runtime\src\libraries\System.Net.Sockets\src\System.Net.Sockets.csproj(8):    <GeneratePlatformNotSupportedAssemblyMessage Condition="'$(TargetsBrowser)' == 'true'">SR.SystemNetSockets_PlatformNotSupported</GeneratePlatformNotSupportedAssemblyMessage>
d:\repos\runtime\src\libraries\System.Net.WebClient\src\System.Net.WebClient.csproj(8):    <GeneratePlatformNotSupportedAssemblyMessage Condition="'$(TargetsBrowser)' == 'true'">SR.SystemNetWebClient_PlatformNotSupported</GeneratePlatformNotSupportedAssemblyMessage>
d:\repos\runtime\src\libraries\System.Security.Cryptography.Algorithms\src\System.Security.Cryptography.Algorithms.csproj(12):    <GeneratePlatformNotSupportedAssemblyMessage Condition="'$(TargetsBrowser)' == 'true'">SR.SystemSecurityCryptographyAlgorithms_PlatformNotSupported</GeneratePlatformNotSupportedAssemblyMessage>
```

Reply:
It depends on how you define "incorrect". Today this doesn't have any impact but that might be different when we bring-up platforms that aren't _represented_ in the project's TargetFrameworks.

Short answer: I can't answer that as I don't have domain knowledge in any of the listed projects. I expect those aren't pure IL ones and need platform support?

Reply:
Most of them have platform-specific code.  WebClient and Requests do not.

Reply:
Do any of them not work without platform support? Those would be candidates for `$(NetCoreAppCurrent)` PNSE configurations.

Reply:
We do use AnyOS-PNSE in a couple other assemblies, just one example: https://github.com/dotnet/runtime/blob/b37f10a504f20faef2558b6d79180411ad0603c3/src/libraries/System.IO.IsolatedStorage/src/System.IO.IsolatedStorage.csproj#L7

I think we just weren't consistent here and to me Viktor's fix feels like the more future proof approach.

Reply:
> I think we just weren't consistent here and to me Viktor's fix feels like the more future proof approach.

In that case, we should switch them all to be the new approach. :wink:

> Do any of them not work without platform support?

All except WebClient/Requests require platform-specific P/Invokes.

Reply:
Do you mind if I file an issue for it to keep the work involved out of this PR?

Reply:
Ok

Reply:
Filed https://github.com/dotnet/runtime/issues/44229

Reply:
Thanks for the feedback :)